pkg/ipmi/ocp/bootorder.go

```diff
@@ -69,64 +72,62 @@ func setBootOrder(i *ipmi.IPMI, BootOrder *BootOrder) error {
 	return nil
 }
 
-// Currently we only support IPv6 Network boot (netboot) and SATA HDD GRUB (localboot),
+// Currently we only support Network boot (pxeboot) and SATA HDD GRUB (boot),
 // other types will be mapped to INVALID_BOOT and put to the end of the bootSeq.
 func remapSortBootOrder(BootOrder *BootOrder) {
-	var bootType byte
 	sorted := [5]byte{INVALID_BOOT, INVALID_BOOT, INVALID_BOOT, INVALID_BOOT, INVALID_BOOT}
 	var idx int
-	for _, v := range BootOrder.bootSeq {
-		bootType = v & 0x7
-		if bootType == NETWORK_BOOT {
-			if v&0x8 == 0 { // If IPv6 first bit is not set, set it
-				v |= 0x8
-			}
-			sorted[idx] = v
-			idx++
-		} else if bootType == LOCAL_BOOT {
-			sorted[idx] = v
+	for _, bootType := range BootOrder.bootSeq {
+		if bootType == NETWORK_BOOT || bootType == NETWORK_BOOT_IPV6 || bootType == LOCAL_BOOT {
+			sorted[idx] = bootType
 			idx++
 		}
 	}
 	copy(BootOrder.bootSeq[:], sorted[:])
 }
 
-func updateVPDBootOrder(i *ipmi.IPMI, BootOrder *BootOrder) error {
+func updateVPDBootOrder(i *ipmi.IPMI, BootOrder *BootOrder, l ulog.Logger) error {
 	var err error
 	var key string
-	var bootType byte
 	var idx int
-	for _, v := range BootOrder.bootSeq {
+	for _, bootType := range BootOrder.bootSeq {
 		key = fmt.Sprintf("Boot%04d", idx)
-		bootType = v & 0x7
 		if bootType == NETWORK_BOOT {
 			log.Printf("VPD set %s to %s", key, NETBOOTER_CONFIG)
 			BootEntries = append(BootEntries, systembooter.BootEntry{Name: key, Config: []byte(NETBOOTER_CONFIG)})
 			if err = vpd.FlashromRWVpdSet(key, []byte(NETBOOTER_CONFIG), false); err != nil {
 				return err
 			}
 			idx++
+		} else if bootType == NETWORK_BOOT_IPV6 {
+			log.Printf("VPD set %s to %s", key, NETBOOTER_IPV6_CONFIG)
```

Comment:
can you remove the print here? This is a package.

Reply:
Done